app_dart/lib/src/request_handling/static_file_handler.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:meta/meta.dart';
+import 'package:mime/mime.dart';
+
+import 'body.dart';
+import 'exceptions.dart';
+
+/// A class based on [RequestHandler] for serving static files.
+@immutable
+class StaticFileHandler<T extends Body> extends RequestHandler<Body> {
+  /// Creates a new [StaticFileHandler].
+  const StaticFileHandler(this.filePath, {Config config}) : super(config: config);
```

Comment:
If you made this class inject a `FileSystem`, defaulting to `const LocalFileSystem()`, then you could mock out the file access in the test.

(see package:file)

Reply:
Thank you! Added